apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -975,6 +1004,25 @@ pub(crate) fn metric_query_planning_plan_duration(planner: &'static str, start:
     );
 }
 
+pub(crate) fn metric_rust_qp_init(init_error_kind: Option<&'static str>) {
+    if let Some(init_error_kind) = init_error_kind {
+        u64_counter!(
+            "apollo.router.lifecycle.query_planner.init",
+            "Rust query planner initialization",
+            1,
+            "init.error_kind" = init_error_kind,
+            "init.is_success" = false
+        );
+    } else {
+        u64_counter!(
+            "apollo.router.lifecycle.query_planner.init",
+            "Rust query planner initialization",
+            1,
+            "init.is_success" = true
+        );
+    }
```

Comment:
Is it ok for a metric attribute to be sometimes missing?

Reply:
I think so. but it could maybe be collapsed to a single attribute (`init.error_kind = "none"`). Though it's the same cardinality either way